### DISCUSSION TITLE
I need help to delete a record inside a field with relationship list

### DISCUSSION THREAD

**Discussion by phantodev** [prisma/prisma#13477] [Q&A] [answered]
Guys... i am on this challange since Monday... please help me? Check my models.

model User {
  id            String    @id @default(cuid())
  bookmarks     Home[]    @relation(name: "UserBookmarks")
}

model Home {
  bookmarkedBy User?    @relation(name: "UserBookmarks", fields: [userId], references: [id])
  userId       String?
}

This example is: ONE PERSON can have MULTIPLE Homes in their BOOKMARKS, ok?

My PATCH it works fine, check above.

`const home = await prisma.home.update({
        where: { id: "123" },
        data: {
          userId: "my-user-id",
        },
});`

When i see the content in PRISMA STUDIO... works fine... i have a list of Homes in bookmarks field. PERFECT.

BUT NOW... how i can DELETE a BOOKMARK in this CASE?

i tried eveything... but i cant understand how i can do that :-( I tried this code bellow:

`const home = await prisma.home.delete({
        where: {
          userId: "my-user-id",
        },
 });`

AND

`const user = await prisma.user.delete({
        where: {
          bookmarks: "my-home-id",
        },
 });`

In this case Above...  appears this message:

{
    "message": "\nInvalid `prisma.home.delete()` invocation:\n\n{\n  where: {\n    userId: 'cl3he5om60008you1geu5hbpa'\n    ~~~~~~\n  }\n}\n\nUnknown arg `userId` in where.userId for type HomeWhereUniqueInput. Did you mean `id`? Available args:\ntype HomeWhereUniqueInput {\n  id?: String\n}\n\n"
}

HELP ME GUYS!

Thanks guys

**Comment by nurul3101** [comment, marked as answer]
Hey @phantodev 👋 

To delete a database record you need to reference it by an id or a unique attribute: [Delete Reference](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#delete)
You are trying to delete it by `userId` which isn't marked as `@unique` in your schema, try adding `@unique` attribute to the schema and then invoking the same query.
Also in your Home model it seems there is no `id` attribute.
Here's a reference schema that you can try
```prisma
model User {
  id        String @id @default(cuid())
  bookmarks Home[] @relation(name: "UserBookmarks")
}

model Home {
  id           String  @id @default(cuid())
  bookmarkedBy User?  @relation(name: "UserBookmarks", fields: [userId], references: [id])
  userId       String? @unique
}

```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
